### PR TITLE
feat: 🎸 Add `trustedFor` method in DefaultTrustedClaimIssuer

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
BREAKING CHANGE: 🧨 
- `TrustedClaimIssuer` interface allows null value for `trustedFor` for specifying all claim types.
- `TrustedClaimIssuers.get` now returns `TrustedClaimIssuer<true>[]`
- `DefaultTrustedClaimIssuer.trustedFor` has been converted to an async method returning claim types for which Claim Issuer is trusted. 
- Allow `null` value for `trustedFor` parameter in
`ModifyTokenTrustedClaimIssuersAddSetParams.claimIssuers` to specify all
claim types for an identity

### CODE REVIEW

**Review by polymath-eric — APPROVED** [review]
LGTM, left a potential nit, but might be OK to leave.

Minor detail in the commit message, but technically "static property" means it's defined on the class itself and not instances. I know what you mean, but it's an overloaded word in the context.

File: src/api/procedures/modifyTokenTrustedClaimIssuers.ts [hunk 1]
```diff
@@ -18,7 +18,10 @@ export interface ModifyTokenTrustedClaimIssuersAddSetParams {
   /**
    * array of Identity IDs
    */
-  claimIssuers: { identity: string | Identity; trustedFor?: ClaimType[] }[];
+  claimIssuers: {
+    identity: string | Identity;
+    trustedFor: ClaimType[] | null;
+  }[];
```
Comment (monitz87) [inline src/api/procedures/modifyTokenTrustedClaimIssuers.ts:24]: This can be `InputTrustedClaimIssuer[]`

**Review by monitz87 — CHANGES_REQUESTED** [review]
You forgot to list the changes to the `TrustedClaimIssuer` interface as breaking changes (and the methods that are affected by the interface change)

File: src/api/entities/DefaultTrustedClaimIssuer.ts [hunk 2]
```diff
@@ -76,4 +70,37 @@ export class DefaultTrustedClaimIssuer extends Identity {
...
+    const claimIssuers = await complianceManager.trustedClaimIssuer(rawTicker);
+
+    const claimIssuer = claimIssuers
+      .map(issuer => trustedIssuerToTrustedClaimIssuer(issuer, context))
+      .find(({ identity: { did: issuerDid } }) => did === issuerDid);
+
+    if (!claimIssuer) {
+      throw new PolymeshError({
+        code: ErrorCode.DataUnavailable,
+        message: `The Identity with DID "${did}" is no longer a trusted issuer for "${ticker}"`,
```
[the remark this answers is not in the document]
Comment (prashantasdeveloper) [reply src/api/entities/DefaultTrustedClaimIssuer.ts:100]: As explained by Jere, I suppose no change is required here.

**Review by polymath-eric — APPROVED** [review]

File: src/types/index.ts [hunk 3]
```diff
@@ -335,10 +336,14 @@ export interface ClaimScope {
...
-  identity: Identity;
+/**
+ * @param IsDefault - whether the Identity is a default trusted claim issuer for an asset or just
+ * for a specific compliance condition. Defaults to false
+ */
+export interface TrustedClaimIssuer<IsDefault extends boolean = false> {
+  identity: IsDefault extends true ? DefaultTrustedClaimIssuer : Identity;
   /**
-   * a null value means that the issuer is trusted for all claim types.
+   * an undefined value means that the issuer is trusted for all claim types.
```
Comment (monitz87) [inline src/types/index.ts:346]:
```suggestion
   * a null value means that the issuer is trusted for all claim types
```

**Review by monitz87 — APPROVED** [review]

**Pull request merged**